packages/components/src/components/post-rating/post-rating.tsx

```diff
@@ -0,0 +1,179 @@
+import { Component, h, Event, EventEmitter, Prop, Host, State, Element } from '@stencil/core';
+import { version } from '../../../package.json';
+
+@Component({
+  tag: 'post-rating',
+  styleUrl: 'post-rating.scss',
+  shadow: true,
+})
+export class PostRating {
+  @Element() host: HTMLPostRatingElement;
+  /**
+   * The current rating value
+   */
+  @Prop({ mutable: true }) currentRating = 0;
+
+  /**
+   * The index of the currently hovered star
+   */
+  @State() hoveredIndex: number;
+
+  /**
+   * The number of stars in the rating
+   */
+  @Prop() readonly max?: number = 5;
+
+  /**
+   * Boolean for the disabled state of the component
+   */
+  @Prop() readonly disabled?: boolean = false;
+
+  /**
+   * If readonly is true, the component only displays a rating and is not interactive.
+   */
+  @Prop() readonly readonly?: boolean = false;
+
+  // prettier-ignore
+  /**
+   * Event emitted when the rating gets commited
+   */
+  @Event({
+    eventName: 'ratingChange',
+    composed: true,
+    bubbles: true,
+  }) ratingChange: EventEmitter<number>;
+
+  // prettier-ignore
+  /**
+   * Event emitted whenever the rating changes
+   */
+  @Event({
+    eventName: 'input',
+    composed: true,
+    bubbles: true,
+  }) input: EventEmitter<number>;
+
+  constructor() {
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+  }
+
+  private handleKeyDown(ev: KeyboardEvent) {
+    switch (ev.key) {
+      case 'ArrowDown':
+      case 'ArrowLeft':
+        ev.preventDefault();
+        this.update(this.currentRating - 1);
+        break;
+      case 'ArrowUp':
+      case 'ArrowRight':
+        ev.preventDefault();
+        this.update(this.currentRating + 1);
+        break;
+      case 'Home':
+        ev.preventDefault();
+        this.update(0);
+        break;
+      case 'End':
+        ev.preventDefault();
+        this.update(this.max);
+        break;
+      case 'Enter':
+      case ' ':
+        this.handleBlur();
+        break;
+      default:
+        return;
+    }
+  }
+
+  private hasChanged = false;
+
+  private handleBlur() {
+    if (this.hasChanged === true) {
```

Comment:
What if it is disabled or readonly? Is it working as expected?

Reply:
I think so, because `hasChanged` can only get `true` if `isInteractive` is true.